uwblueprint/abtc
# Add fetching serviceRequests to calendar view

## Notion ticket link

[Add fetching serviceRequests to calendar view](https://www.notion.so/uwblueprintexecs/Add-fetching-serviceRequests-to-calendar-view-0056327875b544a7bf6e762a58a2d821?pvs=4)



## Implementation description
* Create function to fetch all available valid shifts (i.e. with shiftTime and shiftEndTime defined)
* Display the shifts in the same way displayed on Figma



## Steps to test
1. Change MongoDB database to development (replace `test` with `development` in .env files); it contains previously seeded serviceRequests.
2. Navigate to localhost:3000/calendar
3. Navigate to February to see most of the shifts. Kitchen requests are blue and site requests are green for now
<img width="1440" alt="Screenshot 2024-06-04 at 8 26 24 PM" src="https://github.com/uwblueprint/abtc/assets/88808428/876086fe-e11f-4a4e-bf63-a8650569fbbf">

<img width="1438" alt="Screenshot 2024-06-04 at 8 26 03 PM" src="https://github.com/uwblueprint/abtc/assets/88808428/fc34e5e9-097f-4c7c-8b62-d81229b9f1b0">



## What should reviewers focus on?
* All valid shifts in MongoDB are displayed
* Calendar should be able to be toggled between month and week
* Service requests can be filtered by their type
* Clicking on a day number will open the weekly view for that week


## Checklist
